src/TheThingsNetwork.h

```diff
@@ -20,6 +20,19 @@
 
 #define TTN_BUFFER_SIZE 300
 
+// The Things Products devices
+#if defined(ARDUINO_THINGS_NODE) || defined (ARDUINO_THINGS_UNO)
+typedef HardwareSerial SerialType;
+#define HARDWARE_UART
+
+#elif defined(ARDUINO_ARCH_SAM) || defined(ARDUINO_ARCH_SAMD)
+typedef HardwareSerial SerialType;
+#define HARDWARE_UART
+
+#else
+typedef Stream SerialType;
+#endif
+
```

Comment:
I don't think we should do this in the library. With this code we are not allowing people to use strange setups like a second RN2483 on a SoftwareSerial port on a Things Uno.

And why are we not allowing SoftwareSerial on SAM and SAMD devices?

Reply:
We could also disable it on the Uno... but power usage will increase 
But I can't see a reason why to put a second Rn2483 on the Things Uno

Reply:
But I think no one would put a second RN2843 on the Uno!
My Opinion: make a special constant, you have to define if you want to use a second RN2843!
No beginner would like to connect a second RN2843, the people which think about such things would also be able to make a special configuration! And for the majority of people, the uno will be power efficient.